modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/UpNextQueueImpl.kt

```diff
@@ -232,9 +238,13 @@ class UpNextQueueImpl @Inject constructor(
         episodes.forEach { playLastNow(it, downloadManager, null) }
     }
 
-    override suspend fun removeEpisode(episode: BaseEpisode) {
+    override suspend fun removeEpisode(episode: BaseEpisode, shouldShuffleUpNext: Boolean) {
```

Comment:
There's no need to pass `shouldShuffleUpNext` to this function. You already have access to settings that hold the property you can read. This also simplifies a lot of things because there will be no need to for `RemoveAndShuffle` type since everything can be determined by reading the user setting.